### PR TITLE
Create Proposal Comment API

### PR DESCRIPTION
**Pull request by dternyak**
### What this does

Adds commenting to the Proposal API

A `POST` request to `/<proposal_id>/comments` with the following body will create a comment:

```
{
  "authorId": 1,
  "content": "Wow, I think this is such a great proposal!"
}

```

### What this also does
- Fleshes out Testing framework
- Creates initial proposal test

### What's left
- Add authorization to make sure the user is authorized to add the comment. Waiting on some finalization on EIP-712, so in the meantime there's no authorization.

### CODE REVIEW

**Comment by wbobeirne** [comment]
What's the plan here with `authorId`? Looks like there's no authorization. How will the client divine an ID, and is there a way we can prevent them from spoofing it? It might be good to think about a server-side message signing "challenge" that rewards you with an auth token.

Let me know if you just want to go ahead with the PR as-is for now, we can plan that out later.

**Comment by dternyak** [comment]
Right on @wbobeirne. The plan is still to add verification to the backend, after first signing the message on the client. 

Instructions on how to do that in Python [here](https://ethereum.stackexchange.com/questions/27735/verifying-personal-sign-signature-with-pyethereum).

One thing I'm slightly hung up on is how to structure the API given that, eventually, users may also be commenting via email/account instead (and not web3). There would be a more standard auth system there.

**Comment by dternyak** [comment]
Going to close this PR and resume work once https://github.com/MetaMask/metamask-extension/pull/4803 has been merged. Hopefully we'll soon have mature libraries around a finalized and implemented EIP-712 spec 😁

**Comment by dternyak** [comment]
I managed to fix the mind-numbing issue with the testing framework setup not migrating random tables.

The issue was the sqlite does not seem to play well with alembic when it is being run in-memory. Simply setting a path for it to live on disk solved this.

I also added some additional tests.

File: backend/grant/proposal/views.py [hunk 1]
```diff
@@ -105,4 +134,4 @@ def make_proposal():
         return JSONResponse(message="Proposal with that hash already exists", _statusCode=409)
 
     results = proposal_schema.dump(proposal)
-    return JSONResponse(results, _statusCode=204)
+    return JSONResponse(results, _statusCode=201)
```
Comment (dternyak) [inline backend/grant/proposal/views.py]: Here's a fun story.

204 == "No content"
201 == "Create"

Missing those up leads to headaches.

What happens if you send a JSON body along with a 204 status code? It gets stripped.

The result is that I spend hours debugging flask_testing to try to figure out why JSON parsing fails with a cryptic error. 

The solution is one-line change to the status code.

**Pull request merged**